xdsl/dialects/onnx.py

```diff
@@ -145,7 +145,9 @@ class ElementwiseBinOpBase(IRDLOperation, ABC):
     res = result_def(TensorType[T])
     assembly_format = "`(` $lhs `,` $rhs `)` attr-dict `:` `(` type($lhs) `,` type($rhs) `)` `->` type($res)"
 
-    def __init__(self, lhs: SSAValue, rhs: SSAValue, res_type: Attribute):
+    def __init__(self, lhs: SSAValue, rhs: SSAValue, res_type: Attribute | None = None):
+        if res_type is None:
+            res_type = lhs.type
```

Comment:
I'm not sure if this is necessary/a good idea, the client can just do the same, I'd leave it as-is for now